server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1176,11 +1176,16 @@ public void regenerateCertificatesOf(Entitlement e, boolean ueberCertificate,
     @Override
     @Transactional
     public void regenerateCertificatesOf(String productId, boolean lazy) {
-        List<Pool> poolsForProduct = this.listAvailableEntitlementPools(null, null, null,
-            productId, new Date(), false, false, new PoolFilterBuilder(), null, false, false)
-            .getPageData();
-        for (Pool pool : poolsForProduct) {
-            regenerateCertificatesOf(pool.getEntitlements(), lazy);
+        if (lazy) {
+            poolCurator.markCertificatesDirtyForProductId(productId);
+        }
+        else {
+            List<Pool> poolsForProduct = this.listAvailableEntitlementPools(null, null, null,
+                productId, new Date(), false, false, new PoolFilterBuilder(), null, false, false)
+                .getPageData();
+            for (Pool pool : poolsForProduct) {
+                regenerateCertificatesOf(pool.getEntitlements(), lazy);
```

Comment:
We don't need to pass lazy through here, since we already know it's false at this point.